src/main/src/main/java/org/geoserver/security/DataAccessLimits.java

```diff
@@ -74,11 +69,7 @@ private void writeObject(ObjectOutputStream out) throws IOException {
      */
     protected void writeFilter(Filter filter, ObjectOutputStream out) throws IOException {
         if (filter != null) {
-            try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
-                Encoder encoder = new Encoder(CONFIGURATION);
-                encoder.encode(filter, OGC.Filter, bos);
-                out.writeObject(bos.toByteArray());
-            }
+            out.writeObject(ECQL.toCQL(Collections.singletonList(filter)));
```

Comment:
CQL is known to have round-tripping issues, you might serialize one filter and get back another, that is functionally equivalent, but not syntactically the same.

Reply:
I changed it using the other method that accept a single Filter. is that what you was pointing to? or I should not use the `toCQL` method?

Reply:
Best not to use CQL for filter serialization at all, due to the round-tripping issues I was referring to above.